.github/workflows/ci.yaml

```diff
@@ -30,10 +30,13 @@ jobs:
           key: ${{ runner.os }}-buildx-${{ github.sha }}
           restore-keys: |
             ${{ runner.os }}-buildx-
+      - name: Generate timestamp
```

Comment:
Are we looking to have eexact same timestamps on grid and cuda images? There are separate steps to generate CUDA/GRID timestamp if i'm understanding this right, so I assume they could end up being different due to execution delays, but not sure how much we care about that.

Reply:
It's to differentiate between say one CUDA image from another CUDA image with the same driver version, say 2 min from now. That can happen during different commits to the same branch, that run the CI pipeline. Similar reasoning for the builds too (but gap would usually be a bit longer).

Reply:
Open to alternate ways to do this as well - let me know if you have suggestions. Another option I had considered was the Github run ID, which should be unique. But the number at the end would seem arbitrary, when seen externally.

Reply:
yeah at the end of the day i dont think it matters too much, in case we would need to check whether any given CUDA and GRID images came from the same run, we can just check build logs.

Reply:
yeah, and we could find that build based on the time (vs just build ID).